File: 02/part1.pl
#!/usr/bin/perl
@l=split(/,/,<>);
$l[1]=12;
$l[2]=2;
for($z=0;$z<=$#l;$z+=4) {
  $opcode = $l[$z];
  $a = $l[$l[$z+1]];
  $b = $l[$l[$z+2]];
  $c = $l[$z+3];
  if($opcode == 1) {
    $v = $a + $b;
    $l[$c]=$v;
  } elsif($opcode == 2) {
    $v = $a * $b;
    $l[$c] = $v;
  } elsif($opcode == 99) {
    print $l[0],"\n";
    exit;
 }
}

File: 02/part1.py
#!/usr/bin/python3
import sys
l = sys.stdin.read()
array=l.split(",")
array[1]='12'
array[2]='2'
cnt=0
while array[cnt] != '99':
    opcode = int(array[cnt])
    if opcode == 1:
        a,b,c = [ int(array[int(array[cnt+1])]),
                  int(array[int(array[cnt+2])]),
                  int(array[cnt+3]) ]
        array[c] = str(a+b)
        cnt += 4
    if opcode == 2:
        a,b,c = [ int(array[int(array[cnt+1])]),
                  int(array[int(array[cnt+2])]),
                  int(array[cnt+3]) ]
        array[c] = str(a*b)
        cnt += 4

print(array[0])
